### PR TITLE
Fix java warnings for unchecked conversions in PlayStoreDynamicFeatureManager

### PR DESCRIPTION
**Pull request by GaryQian**
Google3 tests require types be specified. This prevents warnings that are otherwise thrown.

Android linter enforces use of SparseArray when mapping between int and objects.

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/PlayStoreDynamicFeatureManager.java [hunk 1]
```diff
@@ -293,10 +294,10 @@ public void loadDartLibrary(int loadingUnitId, String moduleName) {
     // performant and robust.
 
     // Search directly in APKs first
-    List<String> apkPaths = new ArrayList();
+    List<String> apkPaths = new ArrayList<String>();
```
Comment (xster) [inline shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/PlayStoreDynamicFeatureManager.java:297]: Android supports Java 7 now. You should be able to just do https://www.baeldung.com/java-diamond-operator

e.g. `new ArrayList<>();`

**Review by xster — APPROVED** [review]
Thanks for quick fix

**Comment by GaryQian** [comment]
Waiting on seeing if the revert https://github.com/flutter/engine/pull/22645 lands. If so, I'll merge this into the re-land

File: shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/PlayStoreDynamicFeatureManager.java (hunk 1 above)
Comment (GaryQian) [reply shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/PlayStoreDynamicFeatureManager.java:297]: Yep, though I'll clean this up later, let's get the roll through

**Pull request merged**